validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/ValidatorLoader.java

```diff
@@ -47,14 +44,11 @@
     // Validators map
     for (int i = 0; i < keypairs.size(); i++) {
       BLSKeyPair keypair = keypairs.get(i);
-      int port =
-          Constants.VALIDATOR_CLIENT_PORT_BASE + i + keyProvider.getValidatorPortStartIndex();
-      new ValidatorClient(keypair, port);
-      ManagedChannel channel =
-          ManagedChannelBuilder.forAddress("localhost", port).usePlaintext().build();
+      final LocalMessageSignerService signerService = new LocalMessageSignerService(keypair);
       STDOUT.log(Level.DEBUG, "Validator " + i + ": " + keypair.getPublicKey().toString());
 
-      validators.put(keypair.getPublicKey(), new ValidatorInfo(numNaughtyValidators > 0, channel));
+      validators.put(
+          keypair.getPublicKey(), new ValidatorInfo(numNaughtyValidators > 0, signerService));
```

Comment:
Unrelated but, it might be a good idea to move the naughtyValidator functionality to the test environment.

Reply:
Yeah I think that should just be removed - having the production code be capable of bad behaviour is a bad idea.